File: demo-apps/demojava/app/src/main/java/com/tradehousemedia/tradecore/java/utils/BaseExampleActivity.java
package com.tradehousemedia.tradecore.java.utils;

import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import com.tradehousemedia.tradecore.java.R;

public class BaseExampleActivity extends AppCompatActivity {

    public void setTitle() {
        TextView title = findViewById(R.id.viewTitle);
        title.setText(ExampleHolder.getLastExample().title);
    }

    public LinearLayout getAdContainer() {
        return findViewById(R.id.viewAdHolder);
    }

    public Button getRefreshButton() {
        return findViewById(R.id.viewRefresh);
    }

}

File: apps/demokotlin/app/src/main/java/com/tradehousemedia/tradecore/kotlin/utils/BaseExampleActivity.kt
package com.tradehousemedia.tradecore.kotlin.utils

import android.widget.Button
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.tradehousemedia.tradecore.kotlin.R

open class BaseExampleActivity : AppCompatActivity() {

    fun setTitle() {
        val title = findViewById<TextView>(R.id.viewTitle)
        title.text = ExampleHolder.lastExample.title
    }

    val adContainer by lazy { findViewById<LinearLayout>(R.id.viewAdHolder) }

    val refreshButton by lazy { findViewById<Button>(R.id.viewRefresh) }

}

